### PR TITLE
rimage: add MTL support

### PR DESCRIPTION
**Pull request by abonislawski**

### CODE REVIEW

**Comment by marcinszkudlinski** [comment]
@abonislawski  pls create a draft PR to SOF that uses this rimage commit - we need a CI run using this

File: src/adsp_config.c [hunk 1]
```diff
@@ -1265,6 +1265,129 @@ static int parse_signed_pkg_v2_5(const toml_table_t *toml, struct parse_ctx *pct
 	return 0;
 }
 
+static void dump_signed_pkg_v3_0(const struct signed_pkg_info_ext_v3_0 *signed_pkg)
```
Comment (marcinszkudlinski) [inline src/adsp_config.c:1268]: we can now call it ACE1.5, @dcpleung @lgirdwood @mwasko ?

File: src/manifest.c [hunk 2]
```diff
@@ -1358,6 +1358,151 @@ int man_write_fw_v2_5(struct image *image)
...
+	ri_sha384(image, MAN_META_EXT_OFFSET_V3_0,
+		  sizeof(struct sof_man_adsp_meta_file_ext_v2_5), hash);
+
+	/* hash values in reverse order */
+	for (i = 0; i < SOF_MAN_MOD_SHA384_LEN; i++) {
+		m->signed_pkg.module[0].hash[i] =
+			hash[SOF_MAN_MOD_SHA384_LEN - 1 - i];
+	}
+
+#if 1
```
Comment (lgirdwood) [inline src/manifest.c:1462]: do we still need this, looks like a debug item ?

File: src/include/rimage/manifest.h [hunk 3]
```diff
@@ -105,6 +123,28 @@
 	sizeof(struct sof_man_adsp_meta_file_ext_v1_8) + \
 	MAN_EXT_PADDING)
 
+/*
+ * Firmware manifest header V2.5 used on TGL onwards
```
Comment (lgirdwood) [inline src/include/rimage/manifest.h:127]: Comment looks wrong.

File: src/adsp_config.c (hunk 1 above)
Comment (lgirdwood) [reply src/adsp_config.c:1268]: yeah, I think so - easier to follow e.g. use a ```_ace_v1_5``` suffix.

File: src/manifest.c (hunk 2 above)
Comment (abonislawski) [reply src/manifest.c]: #if removed

**Review by marcinszkudlinski — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@abonislawski did you create a SOF PR to test in the CI ? if so can you link here with it. We can merge once the CI passes. Thanks !

**Comment by abonislawski** [comment]
@lgirdwood we are testing with Marcin right now
https://github.com/thesofproject/sof/pull/5858

**Comment by lgirdwood** [comment]
> @lgirdwood we are testing with Marcin right now
> [thesofproject/sof#5858](https://github.com/thesofproject/sof/pull/5858)

Tests passed.

**Pull request merged**